cmd/river/go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/riverqueue/river/cmd/river
 
 go 1.21.4
```

Comment:
`go mod tidy` wanted to revert some of these submodules back to the `go 1.21.4` specifier. I'm not really sure why and didn't spend much time looking into it, but we can do that if you think it's worthwhile.

Reply:
I think it's okay. One thing I've noticed is that to pick up a toolchain from a version, you need to have the patch version in there. i.e. `go 1.22` wont mind 1.22, but `go 1.22.0` will.